cors/mod.ts

```diff
@@ -0,0 +1,207 @@
+import { Drash } from "../deps.ts";
+import { vary } from "./deps.ts";
+
+interface ArrayOfValueOrArray<T> extends Array<ValueOrArray<T>> {}
+type ValueOrArray<T> = T | ArrayOfValueOrArray<T>;
+type OriginOption =
+  | ((
+    request: Drash.Http.Request,
+    response?: Drash.Http.Response,
+  ) => string | Promise<string>)
+  | ValueOrArray<string | boolean | RegExp>;
+
+type CorsMiddlewareConfig = {
+  origin?: OriginOption;
+  credentials?:
+    | ((
+      request: Drash.Http.Request,
+      response?: Drash.Http.Response,
+    ) => boolean | Promise<boolean>)
+    | boolean;
+  exposeHeaders?: string | string[];
+  allowMethods?: string | string[];
+  allowHeaders?: string | string[];
+  maxAge?: number;
+  optionsSuccessStatus?: number;
+  preflight?: boolean;
+};
+
+const isOriginAllowed = (reqOrigin: string, origin: OriginOption): boolean => {
+  if (Array.isArray(origin)) {
+    for (let i = 0; i < origin.length; ++i) {
+      if (isOriginAllowed(reqOrigin, origin[i])) {
+        return true;
+      }
+    }
+
+    return false;
+  } else if (typeof origin === "string") {
+    return reqOrigin === origin;
+  } else if (origin instanceof RegExp) {
+    return origin.test(reqOrigin);
+  } else {
+    return !!origin;
+  }
+};
+
+export const CorsMiddleware = ({
+  origin = "*",
+  credentials = false,
+  exposeHeaders,
+  allowMethods = "GET,HEAD,PUT,PATCH,POST,DELETE",
+  allowHeaders,
+  maxAge,
+  optionsSuccessStatus = 204,
+  preflight = true,
+}: CorsMiddlewareConfig = {}) =>
+  async (
+    request: Drash.Http.Request,
+    response: Drash.Http.Response,
+  ): Promise<void> => {
+    let options: any = {
+      origin,
+      credentials,
+      optionsSuccessStatus,
+      preflight,
+    };
+
+    if (!options.origin) {
+      return;
+    }
+
+    if (Array.isArray(exposeHeaders)) {
+      options.exposeHeaders = exposeHeaders.join(",");
+    } else {
+      options.exposeHeaders = exposeHeaders;
+    }
+
+    if (Array.isArray(allowMethods)) {
+      options.allowMethods = allowMethods.join(",");
+    } else {
+      options.allowMethods = allowMethods;
+    }
+
+    if (Array.isArray(allowHeaders)) {
+      options.allowHeaders = allowHeaders.join(",");
+    } else {
+      options.allowHeaders = allowHeaders;
+    }
+
+    if (maxAge) {
+      options.maxAge = String(maxAge);
+    }
+
+    // If the Origin header is not present terminate this set of steps.
+    // The request is outside the scope of this specification.
+    const requestOrigin = request.headers.get("Origin");
+
+    // Always set Vary header
+    // https://github.com/rs/cors/issues/10
+    vary(
+      (header: string) => response.headers.get(header) || "",
+      (header: string, value: string) => {
+        response.headers.set(header, value);
+      },
+      "origin",
+    );
+    if (!requestOrigin) {
+      return;
+    }
+
+    let o;
```

Comment:
Can this variable be named so it's descriptive of it's purpose?

Reply:
https://github.com/drashland/deno-drash-middleware/pull/1/commits/1627c064bf674eeeccbe726cd79bd685da78dd8b